### ISSUE TITLE
Make fails on RHEL 6.7 running GOLANG 1.9.2

### ISSUE THREAD

**Issue by DanielNeedles** [google/mtail#119]
It cannot find a depenency

git clone git://github.com/google/mtail
cd mtail
make
::
main.go:17:2: cannot find package "github.com/google/mtail/mtail" in any of:
	/usr/local/go/src/github.com/google/mtail/mtail (from $GOROOT)
	/root/go/src/github.com/google/mtail/mtail (from $GOPATH)
make: *** [mtail] Error 1

ls /root/go/src/github.com/google
   go-cmp

**Comment by DanielNeedles** [comment]
I found the following work around.  After performing the steps in the readme I additionally did the following:

cd /root/go/src/github.com/google/mtail
make
make install

/root/go/bin/mtail -ver
flag provided but not defined: -ver
mtail version v3.0.0-rc4-24-g6c9c925 git revision 6c9c9251cf49f792ecba672369e9a2e1c305508f go version go1.9.2

Usage:
  -address string
    	Host or IP address on which to bind HTTP listener
  -alsologtostderr
    	log to standard error as well as files
::    ::    ::

**Comment by jaqx0r** [comment]
Does the installation guide need some more information about how `$GOPATH` works ?

**Comment by BlessJah** [comment]
@jaqx0r It would be nice. Project is clearly aimed at ops people lacking slightest idea how go works (me included). 

Manual on build process stating that `This assumes that your Go environment is already set up.` might not be enough for people wondering why `make install` is not enough.

**Comment by vgallissot** [comment]
I experience the exact same error and [this issue](https://github.com/google/mtail/issues/91) solved it for me.

TL;DR: you must git clone this repo into your $GOPATH

**Working** example:
```
± docker run -ti golang
root@760e28990298:/go# ls -l src/
total 0
root@760e28990298:/go# mkdir -p src/github.com/google/     
root@760e28990298:/go# cd src/github.com/google/
root@760e28990298:/go/src/github.com/google# git clone https://github.com/google/mtail.git          
Cloning into 'mtail'...
remote: Counting objects: 7242, done.
remote: Compressing objects: 100% (98/98), done.
remote: Total 7242 (delta 172), reused 241 (delta 170), pack-reused 6973
Receiving objects: 100% (7242/7242), 3.36 MiB | 0 bytes/s, done.
Resolving deltas: 100% (4903/4903), done.
root@760e28990298:/go/src/github.com/google# cd mtail
root@760e28990298:/go/src/github.com/google/mtail# make
go get golang.org/x/tools/cmd/goyacc
touch .gen-dep-stamp
go generate -x ./vm
/go/bin/goyacc -v y.output -o parser.go -p mtail parser.y
# Install all dependencies, ensuring they're updated
go get -u -v bufio bytes crypto/rand encoding/json expvar flag fmt github.com/fsnotify/fsnotify github.com/golang/glog github.com/pkg/errors github.com/spf13/afero html/template io io/ioutil math math/big math/rand net net/http net/http/pprof os os/signal path path/filepath regexp regexp/syntax runtime runtime/debug sort strconv strings sync sync/atomic syscall text/tabwriter time unicode unicode/utf8
github.com/fsnotify/fsnotify (download)
Fetching https://golang.org/x/sys/unix?go-get=1
Parsing meta tags from https://golang.org/x/sys/unix?go-get=1 (status code 200)
get "golang.org/x/sys/unix": found meta tag get.metaImport{Prefix:"golang.org/x/sys", VCS:"git", RepoRoot:"https://go.googlesource.com/sys"} at https://golang.org/x/sys/unix?go-get=1
get "golang.org/x/sys/unix": verifying non-authoritative meta tag
Fetching https://golang.org/x/sys?go-get=1
Parsing meta tags from https://golang.org/x/sys?go-get=1 (status code 200)
golang.org/x/sys (download)
github.com/golang/glog (download)
github.com/pkg/errors (download)
github.com/spf13/afero (download)
Fetching https://golang.org/x/text/transform?go-get=1
Parsing meta tags from https://golang.org/x/text/transform?go-get=1 (status code 200)
get "golang.org/x/text/transform": found meta tag get.metaImport{Prefix:"golang.org/x/text", VCS:"git", RepoRoot:"https://go.googlesource.com/text"} at https://golang.org/x/text/transform?go-get=1
get "golang.org/x/text/transform": verifying non-authoritative meta tag
Fetching https://golang.org/x/text?go-get=1
Parsing meta tags from https://golang.org/x/text?go-get=1 (status code 200)
golang.org/x/text (download)
Fetching https://golang.org/x/text/unicode/norm?go-get=1
Parsing meta tags from https://golang.org/x/text/unicode/norm?go-get=1 (status code 200)
get "golang.org/x/text/unicode/norm": found meta tag get.metaImport{Prefix:"golang.org/x/text", VCS:"git", RepoRoot:"https://go.googlesource.com/text"} at https://golang.org/x/text/unicode/norm?go-get=1
get "golang.org/x/text/unicode/norm": verifying non-authoritative meta tag
github.com/golang/glog
github.com/pkg/errors
golang.org/x/sys/unix
github.com/spf13/afero/mem
golang.org/x/text/transform
golang.org/x/text/unicode/norm
github.com/fsnotify/fsnotify
github.com/spf13/afero
go get -u -v bytes encoding/json errors expvar flag fmt github.com/golang/glog github.com/google/go-cmp/cmp github.com/google/go-cmp/cmp/cmpopts github.com/spf13/afero io io/ioutil math/rand net/http net/http/httptest os os/user path path/filepath reflect regexp regexp/syntax runtime sort strconv strings sync syscall testing testing/quick time
github.com/golang/glog (download)
github.com/google/go-cmp (download)
github.com/spf13/afero (download)
Fetching https://golang.org/x/text/transform?go-get=1
Parsing meta tags from https://golang.org/x/text/transform?go-get=1 (status code 200)
get "golang.org/x/text/transform": found meta tag get.metaImport{Prefix:"golang.org/x/text", VCS:"git", RepoRoot:"https://go.googlesource.com/text"} at https://golang.org/x/text/transform?go-get=1
get "golang.org/x/text/transform": verifying non-authoritative meta tag
Fetching https://golang.org/x/text?go-get=1
Parsing meta tags from https://golang.org/x/text?go-get=1 (status code 200)
golang.org/x/text (download)
Fetching https://golang.org/x/text/unicode/norm?go-get=1
Parsing meta tags from https://golang.org/x/text/unicode/norm?go-get=1 (status code 200)
get "golang.org/x/text/unicode/norm": found meta tag get.metaImport{Prefix:"golang.org/x/text", VCS:"git", RepoRoot:"https://go.googlesource.com/text"} at https://golang.org/x/text/unicode/norm?go-get=1
get "golang.org/x/text/unicode/norm": verifying non-authoritative meta tag
github.com/google/go-cmp/cmp/internal/function
github.com/google/go-cmp/cmp/internal/diff
github.com/google/go-cmp/cmp/internal/value
github.com/google/go-cmp/cmp
github.com/google/go-cmp/cmp/cmpopts
touch .dep-stamp
go install -ldflags "-X main.Version=v3.0.0-rc4-126-g470f336 -X main.Revision=470f336b2e05d54c8c30a2af52eb660772113f7e"
root@760e28990298:/go/src/github.com/google/mtail# ls -l /go/bin/mtail 
-rwxr-xr-x 1 root root 9977765 Dec 14 14:58 /go/bin/mtail
root@760e28990298:/go/src/github.com/google/mtail# /go/bin/mtail -V
flag provided but not defined: -V
mtail version v3.0.0-rc4-126-g470f336 git revision 470f336b2e05d54c8c30a2af52eb660772113f7e go version go1.9.2

```


**Not** working:

```
± docker run -ti golang
root@be2e53b76ee5:/go# mkdir -p /usr/local/src/github.com/google/
root@be2e53b76ee5:/go# cd /usr/local/src/github.com/google/
root@be2e53b76ee5:/usr/local/src/github.com/google# git clone https://github.com/google/mtail.git
Cloning into 'mtail'...
remote: Counting objects: 7242, done.
remote: Compressing objects: 100% (98/98), done.
remote: Total 7242 (delta 172), reused 241 (delta 170), pack-reused 6973
Receiving objects: 100% (7242/7242), 3.36 MiB | 3.60 MiB/s, done.
Resolving deltas: 100% (4903/4903), done.
root@be2e53b76ee5:/usr/local/src/github.com/google# cd mtail/
root@be2e53b76ee5:/usr/local/src/github.com/google/mtail# make
go get golang.org/x/tools/cmd/goyacc
touch .gen-dep-stamp
go generate -x ./vm
/go/bin/goyacc -v y.output -o parser.go -p mtail parser.y
# Install all dependencies, ensuring they're updated
go get -u -v bufio bytes crypto/rand encoding/json expvar flag fmt github.com/fsnotify/fsnotify github.com/golang/glog github.com/pkg/errors github.com/spf13/afero html/template io io/ioutil math math/big math/rand net net/http net/http/pprof os os/signal path path/filepath regexp regexp/syntax runtime runtime/debug sort strconv strings sync sync/atomic syscall text/tabwriter time unicode unicode/utf8
github.com/fsnotify/fsnotify (download)
Fetching https://golang.org/x/sys/unix?go-get=1
Parsing meta tags from https://golang.org/x/sys/unix?go-get=1 (status code 200)
get "golang.org/x/sys/unix": found meta tag get.metaImport{Prefix:"golang.org/x/sys", VCS:"git", RepoRoot:"https://go.googlesource.com/sys"} at https://golang.org/x/sys/unix?go-get=1
get "golang.org/x/sys/unix": verifying non-authoritative meta tag
Fetching https://golang.org/x/sys?go-get=1
Parsing meta tags from https://golang.org/x/sys?go-get=1 (status code 200)
golang.org/x/sys (download)
github.com/golang/glog (download)
github.com/pkg/errors (download)
github.com/spf13/afero (download)
Fetching https://golang.org/x/text/transform?go-get=1
Parsing meta tags from https://golang.org/x/text/transform?go-get=1 (status code 200)
get "golang.org/x/text/transform": found meta tag get.metaImport{Prefix:"golang.org/x/text", VCS:"git", RepoRoot:"https://go.googlesource.com/text"} at https://golang.org/x/text/transform?go-get=1
get "golang.org/x/text/transform": verifying non-authoritative meta tag
Fetching https://golang.org/x/text?go-get=1
Parsing meta tags from https://golang.org/x/text?go-get=1 (status code 200)
golang.org/x/text (download)
Fetching https://golang.org/x/text/unicode/norm?go-get=1
Parsing meta tags from https://golang.org/x/text/unicode/norm?go-get=1 (status code 200)
get "golang.org/x/text/unicode/norm": found meta tag get.metaImport{Prefix:"golang.org/x/text", VCS:"git", RepoRoot:"https://go.googlesource.com/text"} at https://golang.org/x/text/unicode/norm?go-get=1
get "golang.org/x/text/unicode/norm": verifying non-authoritative meta tag
golang.org/x/sys/unix
github.com/golang/glog
github.com/pkg/errors
github.com/spf13/afero/mem
golang.org/x/text/transform
golang.org/x/text/unicode/norm
github.com/fsnotify/fsnotify
github.com/spf13/afero
go get -u -v bytes encoding/json errors expvar flag fmt github.com/golang/glog github.com/google/go-cmp/cmp github.com/google/go-cmp/cmp/cmpopts github.com/spf13/afero io io/ioutil math/rand net/http net/http/httptest os os/user path path/filepath reflect regexp regexp/syntax runtime sort strconv strings sync syscall testing testing/quick time
github.com/golang/glog (download)
github.com/google/go-cmp (download)
github.com/spf13/afero (download)
Fetching https://golang.org/x/text/transform?go-get=1
Parsing meta tags from https://golang.org/x/text/transform?go-get=1 (status code 200)
get "golang.org/x/text/transform": found meta tag get.metaImport{Prefix:"golang.org/x/text", VCS:"git", RepoRoot:"https://go.googlesource.com/text"} at https://golang.org/x/text/transform?go-get=1
get "golang.org/x/text/transform": verifying non-authoritative meta tag
Fetching https://golang.org/x/text?go-get=1
Parsing meta tags from https://golang.org/x/text?go-get=1 (status code 200)
golang.org/x/text (download)
Fetching https://golang.org/x/text/unicode/norm?go-get=1
Parsing meta tags from https://golang.org/x/text/unicode/norm?go-get=1 (status code 200)
get "golang.org/x/text/unicode/norm": found meta tag get.metaImport{Prefix:"golang.org/x/text", VCS:"git", RepoRoot:"https://go.googlesource.com/text"} at https://golang.org/x/text/unicode/norm?go-get=1
get "golang.org/x/text/unicode/norm": verifying non-authoritative meta tag
github.com/google/go-cmp/cmp/internal/diff
github.com/google/go-cmp/cmp/internal/value
github.com/google/go-cmp/cmp/internal/function
github.com/google/go-cmp/cmp
github.com/google/go-cmp/cmp/cmpopts
touch .dep-stamp
go install -ldflags "-X main.Version=v3.0.0-rc4-126-g470f336 -X main.Revision=470f336b2e05d54c8c30a2af52eb660772113f7e"
main.go:17:2: cannot find package "github.com/google/mtail/mtail" in any of:
	/usr/local/go/src/github.com/google/mtail/mtail (from $GOROOT)
	/go/src/github.com/google/mtail/mtail (from $GOPATH)
Makefile:78: recipe for target 'mtail' failed
make: *** [mtail] Error 1

```


**env** of this current docker image:
```
± docker run -ti golang env
PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=47e6def07956
TERM=xterm
GOLANG_VERSION=1.9.2
GOPATH=/go
HOME=/root
```

**Comment by jaqx0r** [comment]
This is the same problem as reported in #62 and #91 , and again, I don't think this project is a substitute for reading the Go docs, for example:
https://github.com/golang/go/wiki/GOPATH#repository-integration-and-creating-go-gettable-projects
https://github.com/golang/go/wiki/SettingGOPATH

https://github.com/alco/gostart

https://golang.org/doc/code.html

**Comment by DanielNeedles** [comment]
I do suspect this was user error. However, I did spend a few hours trying to determine right approach and it ended up not being Googlefoo or RTFM accessible.  I think a few well placed statements in the readme is called for.  That said, it is a free Internet and you’re welcome to do whatever you want.  8-)  It was enough of a hurdle that I found a different solution.

**Comment by DanielNeedles** [comment]
To be honest I moved to a different solution. I do suspect it was simply being a newbee trying to find my way. That said, it DEFINITELY wasn’t a normal issue that I’ve come across with other languages.  There should be a note in the readme on how to install and deploy if it isn’t intuitive.   After several hours of Google Foo I threw in the towel and went with a different approach that is working for me now.

**Comment by jaqx0r** [comment]
I'm sorry it was difficult. You've convinced me that the very least can do
is to prompt new users to the correct documentation and do better for first
time installers.

**Comment by DanielNeedles** [comment]
Thanks!  That would be great.  Sorry for the bit of “snarkiness.”  Yesterday was a bad day.  8-)

**Comment by jaqx0r** [comment]
Fixed with 8650c21ade456c06b018e1a76338e2b0413838b6
